src/Servers/IIS/AspNetCoreModuleV2/RequestHandlerLib/requesthandler_config.cpp

```diff
@@ -248,12 +251,43 @@ REQUESTHANDLER_CONFIG::Populate(
         goto Finished;
     }
 
-    hr = GetElementStringProperty(pAspNetCoreElement,
-        CS_ASPNETCORE_PROCESS_EXE_PATH,
-        &m_struProcessPath);
-    if (FAILED(hr))
+    // We prefer the environment variables for LAUNCHER_PATH and LAUNCHER_ARGS
+    if (launcherPathEnv.has_value())
     {
-        goto Finished;
+        hr = m_struProcessPath.Copy(launcherPathEnv.value().c_str());
+        if (FAILED(hr))
```

Comment:
nit: we have macros now for FINISHED_IF_FAILED which could clean this up a good amount.